api/src/main/java/org/openmrs/module/fhir/api/util/FHIRObsUtil.java

```diff
@@ -240,4 +241,51 @@ public static Observation generateObs(Obs obs) {
 		}
 		return observation;
 	}
+	
+	public static Obs generateOpenMRSObs(Observation observation, List<String> errors) {
+		Obs obs = new Obs();
+		
+		obs.setComment(observation.getComments());
+		if (observation.getSubject() != null) {
+			ResourceReferenceDt subjectref = observation.getSubject();
+			IdDt id = subjectref.getReference();
+			String uri = id.getValue();
+			String[] paths = uri.split("/");
+			String patientUuid = paths[1];
+			obs.setPerson(Context.getPersonService().getPersonByUuid(patientUuid));
+		} else {
+			errors.add("Subject cannot be null");
```

Comment:
I also thought about that. I get this from location resource. In a way, this is good. we accumulate all the erros, then at the serviceImpl method, we check whether the error list is empty or not. If it is not empty we populate ALL error methods to the end user and return unprocessableentitiy exception. 

This way the user get all the errror messages in his request. What do you think?